### DISCUSSION TITLE
Support "top_of_page_button": "blob" to avoid requiring sign-in

### DISCUSSION THREAD

**Discussion by umarcor** [pradyunsg/furo#630] [Potential Issue] [answered]
Currently, the optional "pencil" button at the top of the page can only be set to either `edit` or `None` (see https://pradyunsg.me/furo/customisation/#top-of-page-button).

On GitHub, accessing the edit link requires being signed in. However, the equivalent `blob` URL can be browsed without signing in. See, for instance, https://github.com/pradyunsg/furo/edit/main/docs/quickstart.md and https://github.com/pradyunsg/furo/blob/main/docs/quickstart.md. Users who are already signed in can go from blob to edit through a single click. Hence, I typically prefer my docs to point to the blob URL.

I looked at the code and it seems non-trivial to change, since this is hardcoded in a dependecy:

- https://github.com/pradyunsg/furo/blob/main/src/furo/theme/furo/components/edit-this-page.html#L2
  - https://github.com/pradyunsg/sphinx-basic-ng/blob/main/src/sphinx_basic_ng/theme/basic-ng/components/edit-this-page.html#L30-L33

For completeness, it can be worked around through the custom edit button (see https://pradyunsg.me/furo/customisation/edit-button/#with-arbitrary-urls). However, AFAIAA the icon would remain the same.

I believe that making docs generally accessible to the general public is a legit use case to consider supporting this feature through `source_repository`, `source_branch` and `source_directory`. @pradyunsg would you be open to supporting this in sphinx-basig-ng?

Refs:

- https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html#confval-vcs_pageview_mode
- https://github.com/pradyunsg/furo/discussions/374

**Comment by pradyunsg** [comment, marked as answer]
Same as https://github.com/pradyunsg/furo/issues/478
